docs/DeveloperGuide.md

```diff
@@ -1,38 +1,335 @@
 # Developer Guide
 
```

Comment:
Very comprehensive sequence diagrams that show the flow of the application well.